### PR TITLE
feat: Let operators pick stackable version automatically

### PR DESCRIPTION
**Pull request by sbernauer**
## Description

Part of https://github.com/stackabletech/issues/issues/404

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Changed
+
+- [BREAKING] ProductImageSelection now defaults `stackableVersion` to operator version ([#619]).
```
Comment (soenkeliebau) [inline CHANGELOG.md:9]: Is this really breaking if the field was mandatory before? 
For all existing and working crds behavior should be unchanged if I am not mistaken?

Comment (sbernauer) [reply CHANGELOG.md:9]: its not necessarily breaking for the end user, but for the clients that use operator-rs (our operators).
From like a Rust code perspective ;)

Comment (maltesander) [reply CHANGELOG.md:9]: Yeah meaning "you cannot just update the operator-rs version...". Maybe should use CRD_BREAKING and OP_BREAKING or sth like this?

**Review by maltesander — APPROVED** [review]
LGTM! Just the changelog comment.

**Review by soenkeliebau — DISMISSED** [review]
Had a discussion with @nightkr about this today and she made some very good points in favor of "version operators and images in lockstep" :)

**Comment by sbernauer** [comment]
It would be great if you could add those to https://github.com/stackabletech/documentation/pull/425

File: src/commons/product_image_selection.rs [hunk 2]
```diff
@@ -70,55 +73,67 @@ pub struct ResolvedProductImage {
...
-                let repo = stackable_version
+            ProductImageSelection::StackableVersion(image_selection) => {
+                let repo = image_selection
                     .repo
                     .as_deref()
                     .unwrap_or(STACKABLE_DOCKER_REPO);
+                let stackable_version = image_selection
+                    .stackable_version
+                    .clone()
+                    .unwrap_or(operator_version.to_string());
```
Comment (maltesander) [inline src/commons/product_image_selection.rs:121]:
```suggestion
                let stackable_version = image_selection
                    .stackable_version
                    .as_deref()
                    .unwrap_or(operator_version);
```

**Review by maltesander — APPROVED** [review]
One minor thing. LGTM otherwise.

File: src/commons/product_image_selection.rs (hunk 2 above)
Comment (sbernauer) [reply src/commons/product_image_selection.rs:121]: I changed this like 3 times and switched back and forth xD Thx!

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**